src/lib.rs

```diff
@@ -0,0 +1,48 @@
+#![deny(missing_docs)]
+
+//! MC compliance check
+
+/// Compliance
+mod common;
+
+pub use common::{ConfigError, Location, LocationProvider};
+
+mod ip_info;
+mod ip_who;
```

Comment:
Let's make the `mod`s conditional, too.

```suggestion
#[cfg(feature = "ip_info_provider")]
mod ip_info;
#[cfg(feature = "ip_who_provider")]
mod ip_who;
```